project-base/src/Controller/Front/NewsletterController.php

```diff
@@ -53,12 +64,43 @@ public function subscribeEmailAction(Request $request)
         $form = $this->createSubscriptionForm();
         $form->handleRequest($request);
 
-        if ($form->isSubmitted() && $form->isValid()) {
-            $email = $form->getData()['email'];
-            $this->newsletterFacade->addSubscribedEmail($email, $this->domain->getId());
+        if ($form->isSubmitted()) {
+            $this->formTimeProvider->generateFormTime($form->getName());
+
+            if ($form->isValid()) {
+                $email = $form->getData()['email'];
+
+                try {
+                    $this->newsletterFacade->addSubscribedEmail($email, $this->domain->getId());
+                    return $this->json(['success' => true]);
+                } catch (EmailWasAlreadySubscribedException $exception) {
+                    return $this->json([
+                        'success' => false,
+                        'errors' => [t('Email was already subscribed')],
```

Comment:
can you dump translations for this PR?

Reply:
288caf2bc38388cab13abf12283b6df44f5f6a2d